mc-mark-more.el

```diff
@@ -126,12 +166,18 @@ Use like case-fold-search, don't recommend setting it globally.")
        (when skip-last
          (mc/remove-fake-cursor furthest-cursor))
        (if (funcall search-function re nil t)
-           (progn
-             (push-mark (funcall match-point-getter 0))
-             (when point-out-of-order
-               (exchange-point-and-mark))
-             (mc/create-fake-cursor-at-point))
-         (user-error "no more matches found."))))))
+           (mc--mark-more-like-this)
+         (when (mc/handle-more-like-this-no-match-condition "No more matches found.")
+	   (goto-char end-char)
```

Comment:
Can't we instead jump to `point-min`? I don't see any logical way there could be a cursor before from where we started, so we can just start from the beginning.  We can set the `nil` argument in `search-function` to the limit (the point where the first cursor was created).  Then we can probably get rid of the repeated `funcall` below.

Reply:
If we only ever goto point min we will keep adding cursors to the very first looped-result. We need to start searching from the last cursor to the current cursor. Maybe sending a BOUND to `search-function`, would work for that instead of the logic I have involving `right-bound` (which seems to fail on prev). I've updated the logic to get rid of `(mc--mark-more-like-this)`.

Reply:
Only using point-min would fail. Every time we run this command we start with the point and mark at the currently selected region. This means that we need to find the most recently created cursor so that we know where to start matching again. Originally we would use the cursor that was the furthest way. With looping it is more complicated. We want to only use the cursor that is furthest away if there is another match after that. If there is no match then we want to find the nearest one to our current point before the point so that we can start searching and placing from there.

My newest change is essentially only changing the search function to place the point at the most recently created cursor or at point min if there are no "before" cursors

Reply:
I think I understand what you mean when I consider that we can skip matches when adding more cursors.

Do you also consider the case when user might have called `mc/cycle-forward`?